tests/jballerina-unit-test/src/test/resources/test-src/types/finite/finite_type_negative.bal

```diff
@@ -118,3 +118,39 @@ function testFiniteTypeWithNumericValues() {
     Foo4 f5 = 3;
     "chiran" x = 5;
 }
+
+type IntOrNull int|null;
+type IntOrNullStr int|"null";
+
+function testNullFiniteType() {
+    IntOrNull a = 1; // OK
+    IntOrNullStr b = 1; // OK
+
+    int|null c = 1;
+    int|"null" d = 1;
+    null e = null;
+    "null" f = "null";
+
+    IntOrNull _ = null; // OK
+    IntOrNull _ = (); // OK
+    IntOrNull _ = "null"; // error
+    IntOrNull _ = a; // OK
+    IntOrNull _ = b; // error
+    IntOrNull _ = c; // OK
+    IntOrNull _ = d; // error
+    IntOrNull _ = e; // OK
+    IntOrNull _ = f; // error
+
+    IntOrNullStr _ = null; // error
+    IntOrNullStr _ = (); // error
+    IntOrNullStr _ = "null"; // OK
+    IntOrNullStr _ = a; // error
+    IntOrNullStr _ = b; // OK
+    IntOrNullStr _ = c; // error
+    IntOrNullStr _ = d; // OK
+    IntOrNullStr _ = e; // error
+    IntOrNullStr _ = f; // OK
+
+    null _ = "null"; // error
+    "null" _ = null; // error
```

Comment:
Shall we replicate these tests in the bala test also?

Reply:
Added